### DISCUSSION TITLE
How can I inject IQueryManager into a custom service?

### DISCUSSION THREAD

**Discussion by kharaei** [OrchardCMS/OrchardCore#16699] [Q&A]
Hello

When I inject IQueryManager into the controller and use it in one of the actions, the project runs without any issues. However, when I define a service to inject it into different controllers, the following error occurs: "Unable to resolve service for type 'OrchardCore.Queries.IQueryManager' while attempting to activate 'OrchardCms.Services.UserService'." 
The interface and the implementation class of the service are in the same project where the controllers exist, and the OrchardCore.Queries package is installed.

Could you please guide me?

**Comment by hishamco** [comment]
Does `OrchardCore.Queries` is enabled? Which OC version?

**Comment by kharaei** [comment]
_Does OrchardCore.Queries is enabled?_ 
**Yes**

_Which OC version?_
**1.8.3**

**Reply by hishamco** [reply]
Could you please try to reproduce the issue in `2.0.0` otherwise I will check before I update to `2.0.0`

**Comment by weirdyang** [comment]
I can't seem to reproduce this, the only time I got the same error was when the queries feature isn't enabled.

Do you mind sharing your code?
